### PR TITLE
[0.2.2 milestone] find_coords: allow 4D input

### LINKED ISSUE #922: find_xyz_cut_coords crashes on some compact binary images
...trace back is:

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-44-802cef648458> in <module>()
----> 1 plotting.find_xyz_cut_coords(nib.load(roi_paths[2]))

/git/nilearn/nilearn/plotting/find_cuts.py in find_xyz_cut_coords(img, mask, activation_threshold)
     90         return .5 * np.array(data.shape)
     91     mask = largest_connected_component(mask)
---> 92     slice_x, slice_y, slice_z = ndimage.find_objects(mask)[0]
     93     my_map = my_map[slice_x, slice_y, slice_z]
     94     mask = mask[slice_x, slice_y, slice_z]
```

The image on which this crashes is attached.

[dmpfc-3.nii.zip](https://github.com/nilearn/nilearn/files/66476/dmpfc-3.nii.zip)

### PR DESCRIPTION
**Pull request by banilo**
tries to fix #922

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
The best solution would probably be to insert a 'check_niimg3d' at the beginning of the function. It should solve the problem and it would be more robust (because it would enable working with filenames).

Also, codacy is unhappy because you have introduced trailing whitespace.

**Comment by GaelVaroquaux** [comment]
Also, (and unrelated to the bug mentioned), we should probably be using a '_safe_get_data' to avoid side effects that are very hard to diagnose and do things like screwing up caching.

**Comment by banilo** [comment]
All comments addressed.

File: nilearn/plotting/tests/test_find_cuts.py [hunk 1]
```diff
@@ -42,6 +42,14 @@ def test_find_cut_coords():
...
         0.5 * np.array(data.shape).astype(np.float))
 
+    # regression test (cf. #922)
+    # pseudo-4D images as input (i.e., X, Y, Z, 1)
+    # previously raised "ValueError: too many values to unpack"
+    data = np.ones((36, 43, 36))[..., np.newaxis]
+    affine = np.eye(4)
+    img = nibabel.Nifti1Image(data, affine)
+    x, y, z = find_xyz_cut_coords(img, activation_threshold=1.1)
+
```
Comment (bthirion) [inline nilearn/plotting/tests/test_find_cuts.py]: You don't test the output ?
x, y, z is supposed to be what ? To me the result is empty for activation_threshold=1.1

**Comment by GaelVaroquaux** [comment]
LGTM. + 1 for merge

File: nilearn/plotting/tests/test_find_cuts.py (hunk 1 above)
Comment (AlexandreAbraham) [inline nilearn/plotting/tests/test_find_cuts.py]: This checks if no exception is raised.

Comment (lesteve) [inline nilearn/plotting/tests/test_find_cuts.py]: For a slightly better test, You could do something like this:

```python
rng = np.random.RandomState(0)
data_3d = rng.randn((10, 10, 10))
data_4d = data_3d[..., np.newaxis]
affine = np.eye(4)
img_3d = nibabel.Nifti1Image(data_3d, affine)
img_4d = nibabel.Nifti1Image(data_4d, affine)
assert_equal(find_xyz_cut_coords(img_3d), find_xyz_cut_coords(img_4d))
```

Comment (banilo) [inline nilearn/plotting/tests/test_find_cuts.py]: Ok, thanks good idea!

2016-01-12 11:26 GMT+01:00 Loïc Estève <notifications@github.com>:

**Comment by lesteve** [comment]
LGTM, the AppVeyor failure looks like a glitch to me. Merging, thanks !

**Pull request merged**